x-pack/plugins/lens/public/editor_frame_service/editor_frame/config_panel/dimension_container.tsx

```diff
@@ -7,7 +7,11 @@
 
 import './dimension_container.scss';
```

Comment:
Hey @patrykkopycinski I checked this PR as we talked async and to me it seems like we should probably think about solution for `EuiOutsideClickDetector` - it doesn't work correctly in any place with a portal element inside `EuiOutsideClickDetector` but maybe changing the implementation would help out here?

@thompsongl what do you think? (please watch the video below with sound)

here's the details of the problem: Patryk tries to update react, but when updating, the dimension panel in lens closes after making an edit in some cases. So I try to edit the field (which is a combobox inside the dimension panel) and then after the edition it's being closed. It's being closed, because we keep the content of Dimension Editor inside `EuiOutsideClickDetector`and with react changed, it detects the combobox content as outside click. 

here's the video with explanation (I am working on this PR, but reverted changes for this file only as they don't make much sense atm) 

https://user-images.githubusercontent.com/4283304/166441330-c6aedc62-c6e7-4f62-b31e-52becc1ce2fe.mov

Reply:
Hey @mbondyra 

I put together a simplified repro and it looks like `EuiOutsideClickDetector` still works with portals: https://codesandbox.io/s/ve1xz2?file=/demo.js

It may be something else with the composition of that panel, but it's difficult to tell. However, looking at the code I do have a suggestion that should help: use `EuiFocusTrap` for everything.

```diff
   return isOpen ? (
     <div ref={panelRef}>
-      <EuiFocusTrap disabled={!focusTrapIsEnabled} clickOutsideDisables={true}>
+      <EuiFocusTrap
+        disabled={!focusTrapIsEnabled}
+        clickOutsideDisables={true}
+        onClickOutside={(event) => {
+          if (isFullscreen || fromExcludedClickTarget(event)) {
+            return;
+          }
+          closeFlyout();
+        }}
+      >
         <EuiWindowEvent event="keydown" handler={closeOnEscape} />
-        <EuiOutsideClickDetector
-          onOutsideClick={(event) => {
-            if (isFullscreen || fromExcludedClickTarget(event)) {
-              return;
-            }
-            closeFlyout();
-          }}
-          isDisabled={!isOpen}
-        >
           <div
             role="dialog"
             aria-labelledby="lnsDimensionContainerTitle"
```